app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt

```diff
@@ -131,6 +150,15 @@ class ChatRoomAdapter(
         }
     }
 
+    private fun shouldGroupMessage(a: Message, b: Message): Boolean {
+        if (a.senderAlias == b.senderAlias && a.sender?.id == b.sender?.id && !a.isSystemMessage() && !b.isSystemMessage()) {
+            val date1 = a.getDate()
+            val date2 = b.getDate()
+            return date1.isSameDay(date2) && date1.timeInMillis - date2.timeInMillis < messageGroupingPeriod * 1000
```

Comment:
its an edge case but, if I send a message at 11:59 PM and then another at 12:01 AM on the next day, I would expect these to group together, but this would not occur given the above logic. Is there a reason for the isSameDay check?

Reply:
Yes It will not group message with different days. In the upstream issue it is mentioned that "When there's a day separator, can't be sequential", so logic is based on that.

If you want to group message even if different day then we can remove "isSameDay" condition. But that will not look good because there will be day separator.

What do you think?

Reply:
fwiw I think as a user I would actually expect/want to see the separator upon the day change, so I don't see the need for getting grouping to work in that case